test/Microsoft.AspNetCore.Http.Abstractions.Tests/EndpointHttpContextExtensionsTests.cs

```diff
@@ -0,0 +1,140 @@
+        public void GetEndpoint_ContextWithFeatureAndNullEndpoint_ReturnsNull()
+        {
+            // Arrange
+            var context = new DefaultHttpContext();
+            context.Features.Set<IEndpointFeature>(new EndpointFeature
+            {
+                Endpoint = null
+            });
+
+            // Act
+            var endpoint = context.GetEndpoint();
+
+            // Assert
+            Assert.Null(endpoint);
+        }
+
+        [Fact]
+        public void GetEndpoint_ContextWithFeatureAndEndpoint_ReturnsNull()
+        {
+            // Arrange
+            var context = new DefaultHttpContext();
+            var initial = new Endpoint(c => Task.CompletedTask, EndpointMetadataCollection.Empty, "Test endpoint");
+            context.Features.Set<IEndpointFeature>(new EndpointFeature
+            {
+                Endpoint = initial
+            });
+
+            // Act
+            var endpoint = context.GetEndpoint();
+
+            // Assert
+            Assert.Equal(initial, endpoint);
+        }
+
+        [Fact]
+        public void SetEndpoint_NullOnContextWithoutFeature_NoFeatureSet()
+        {
+            // Arrange
+            var context = new DefaultHttpContext();
+
+            // Act
+            context.SetEndpoint(null);
+
+            // Assert
+            Assert.Null(context.Features.Get<IEndpointFeature>());
+        }
+
+        [Fact]
+        public void SetEndpoint_EndpointOnContextWithoutFeature_FeatureWithEndpointSet()
+        {
+            // Arrange
+            var context = new DefaultHttpContext();
+
+            // Act
+            var endpoint = new Endpoint(c => Task.CompletedTask, EndpointMetadataCollection.Empty, "Test endpoint");
+            context.SetEndpoint(endpoint);
+
+            // Assert
+            var feature = context.Features.Get<IEndpointFeature>();
+            Assert.NotNull(feature);
+            Assert.Equal(endpoint, feature.Endpoint);
+        }
+
+        [Fact]
+        public void SetEndpoint_EndpointOnContextWithFeature_EndpointSetOnExistingFeature()
+        {
+            // Arrange
+            var context = new DefaultHttpContext();
+            var initialEndpoint = new Endpoint(c => Task.CompletedTask, EndpointMetadataCollection.Empty, "Test endpoint");
+            var initialFeature = new EndpointFeature
+            {
+                Endpoint = initialEndpoint
+            };
+            context.Features.Set<IEndpointFeature>(initialFeature);
+
+            // Act
+            var endpoint = new Endpoint(c => Task.CompletedTask, EndpointMetadataCollection.Empty, "Test endpoint");
+            context.SetEndpoint(endpoint);
+
+            // Assert
+            var feature = context.Features.Get<IEndpointFeature>();
+            Assert.Equal(initialFeature, feature);
+            Assert.Equal(endpoint, feature.Endpoint);
+        }
+
+        [Fact]
+        public void SetEndpoint_NullOnContextWithFeature_NullSetOnExistingFeature()
```

Comment:
You never actually call SetEndpoint and GetEndpoint in the same test to ensure it round trips.